sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs

```diff
@@ -164,51 +193,113 @@ public async Task GetTokenForTeamsUserWithEmptyTokenShouldThrow()
             }
-            catch (ArgumentNullException ex)
+            catch (RequestFailedException ex)
             {
-                Assert.AreEqual("token", ex.ParamName);
+                Assert.NotNull(ex.Message);
+                Assert.True(ex.Message.Contains("401"));
+                Console.WriteLine(ex.Message);
                 return;
             }
             Assert.Fail("An exception should have been thrown.");
         }
 
         [Test]
-        public async Task GetTokenForTeamsUserWithInvalidTokenShouldThrow()
+        [TestCase("", TestName = "GetTokenForTeamsUserWithEmptyClientIdShouldThrow")]
+        [TestCase("invalid", TestName = "GetTokenForTeamsUserWithInvalidClientIdShouldThrow")]
+        public async Task GetTokenForTeamsUserWithInvalidClientIdShouldThrow(string clientId)
         {
+            if (TestEnvironment.ShouldIgnoreIdentityExchangeTokenTest)
+            {
+                Assert.Ignore("Ignore exchange teams token test if flag is enabled.");
+            }
+
             try
             {
                 CommunicationIdentityClient client = CreateClientWithConnectionString();
-                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync("invalid");
+                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync(new GetTokenForTeamsUserOptions(CTEOptions.TeamsUserAadToken, clientId, CTEOptions.UserObjectId));
             }
             catch (RequestFailedException ex)
             {
                 Assert.NotNull(ex.Message);
-                Assert.True(ex.Message.Contains("401"));
+                Assert.True(ex.Message.Contains("400"));
                 Console.WriteLine(ex.Message);
                 return;
             }
             Assert.Fail("An exception should have been thrown.");
         }
 
         [Test]
-        public async Task GetTokenForTeamsUserWithExpiredTokenShouldThrow()
+        public async Task GetTokenForTeamsUserWithWrongClientIdShouldThrow()
         {
+            if (TestEnvironment.ShouldIgnoreIdentityExchangeTokenTest)
+            {
+                Assert.Ignore("Ignore exchange teams token test if flag is enabled.");
+            }
+
             try
             {
                 CommunicationIdentityClient client = CreateClientWithConnectionString();
-                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync(TestEnvironment.CommunicationExpiredTeamsToken);
+                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync(new GetTokenForTeamsUserOptions(CTEOptions.TeamsUserAadToken, CTEOptions.UserObjectId, CTEOptions.UserObjectId));
             }
             catch (RequestFailedException ex)
             {
                 Assert.NotNull(ex.Message);
-                Assert.True(ex.Message.Contains("401"));
+                Assert.True(ex.Message.Contains("400"));
```

Comment:
btw how come this has changed from 401 to 400 @paveldostalms @AikoBB  ? is it correct?

Reply:
yep, for the new API version, the service responds with 400 status code for invalid params. Correct me if I am wrong @paveldostalms

Reply:
Yes, invalid clientId or userId is causing 400 as requested by PM. Invalid token continue to return 401.